internal/session/repository_session.go

```diff
@@ -541,6 +581,11 @@ func (r *Repository) ActivateSession(ctx context.Context, sessionId string, sess
 		},
 	)
 	if err != nil {
+		// If this was a duplicate activation attempt, return existing session if the tofu token matches
+		if strings.Contains(err.Error(), "duplicate key value violates unique constraint") {
+			event.WriteSysEvent(ctx, op, fmt.Sprintf("Ignoring duplicate session activation attempt for session %v", sessionId))
+			return r.getActivatedSession(ctx, sessionId, tofuToken)
```

Comment:
The issue is with the activateStateCte (I'm able to reproduce this reliably), which causes it to error our before the tofu token check: https://github.com/hashicorp/boundary/blob/e80316cc60602afca594c5b8347ec56c66563a38/internal/session/repository_session.go#L537-L542

Since that fails the transaction, that's why I have the check here for the dupe error

Reply:
Seems like we need to fix the CTE!